### PR TITLE
feat: add token from dapp

### PR DESCRIPTION
**Pull request by janek26**
As a DApp Developer I want to be able to add my token to the wallet extension programmatically.

### CODE REVIEW

File: packages/extension/src/inpage/index.ts [hunk 1]
```diff
@@ -24,12 +24,75 @@ function sendMessage(msg: MessageType): void {
...
+}
+
 // window.ethereum like
 const starknetWindowObject: StarknetWindowObject = {
   signer: undefined,
   provider: defaultProvider,
   selectedAddress: undefined,
   isConnected: false,
+  request: (call) =>
+    new Promise(async (resolve, reject) => {
```
Comment (delaaxe) [inline packages/extension/src/inpage/index.ts:57]: No need to use `new Promise` here

File: packages/extension/src/inpage/model.ts [hunk 2]
```diff
@@ -2,7 +2,26 @@ import type { Provider, SignerInterface } from "starknet"
...
+  options: {
+    address: string // The hexadecimal StarkNet address of the token contract
+    symbol?: string // A ticker symbol or shorthand, up to 5 alphanumerical characters
+    decimals?: number // The number of asset decimals
+    image?: string // A string url of the token logo
+    name?: string // The name of the token - not in spec
+  }
+}
+
+export type RpcMessage = {
```
Comment (delaaxe) [inline packages/extension/src/inpage/model.ts:18]: Interface?

**Review by delaaxe — APPROVED** [review]

File: packages/extension/src/inpage/model.ts (hunk 2 above)
Comment (janek26) [reply packages/extension/src/inpage/model.ts:18]: yes and no, normally I would prefer an interface, but when adding more RpcMessages we need to go back to a type, as it's an enum, ie:
```
export type RpcMessage = {
  type: "rpcMethod1",
  data: ...,
} | {
  type: "rpcMethod2",
  data ...,
}
```

Comment (delaaxe) [reply packages/extension/src/inpage/model.ts:18]: I know that was the intention but I would only switch from `interface` to `union` once it's actually the case, so no premature optimization.

Those things become obvious once you integrate a linter in your workflow.

**Pull request merged**